spec/requests/posts_spec.rb

```diff
@@ -1,7 +1,27 @@
 require 'rails_helper'
 
 RSpec.describe 'Posts', type: :request do
-  describe 'GET /index' do
-    pending "add some examples (or delete) #{__FILE__}"
+  context 'GET /index' do
+    let(:valid_user_id) { 1 } # Replace with a valid user ID
+
+    before :each do
+      get "/users/#{valid_user_id}/posts"
+    end
+
+    it 'returns a successful response' do
+      expect(response).to be_successful
+    end
+
+    it 'returns HTTP status 200' do
+      expect(response.status).to eq(200)
+    end
+
+    it 'renders the right view file' do
+      expect(response).to render_template('posts/index')
+    end
+
+    it 'renders the right placeholder' do
+      expect(response.body).to include('<h1>Here is a list of posts for a given user</h1>')
+    end
```

Comment:
- [ ] Well done for testing the `index` action of `Post` controller ✨👏 Kindly note that we need to test each action of our controllers as per [this project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Create%20a%20Request,action%20you%20check%3A), therefore, please make sure to test the `show` action of the `Post` controller as well ⛏